src/core/store/modules/notifications/notifications.reducer.ts

```diff
@@ -0,0 +1,42 @@
+import { createReducer } from '@reduxjs/toolkit';
+
+import { initialStatus, NotificationsState } from '@types';
+
+import { NotificationsActions } from './notifications.actions';
+
+export const notificationsInitialState: NotificationsState = {
+  messages: [],
+  dismissedMessageIds: [],
+  statusMap: {
+    getNotificationMessages: { ...initialStatus },
+  },
+};
+
+const { getNotificationMessages, dismissMessage } = NotificationsActions;
+
+const notificationsReducer = createReducer(notificationsInitialState, (builder) => {
+  builder
+    .addCase(dismissMessage, (state, actions) => {
+      state.dismissedMessageIds = [...state.dismissedMessageIds, actions.payload];
+    })
+
+    /* -------------------------------------------------------------------------- */
+    /*                                 Fetch Data                                 */
+    /* -------------------------------------------------------------------------- */
+    .addCase(getNotificationMessages.pending, (state) => {
+      state.statusMap.getNotificationMessages = { loading: true };
+    })
+
+    .addCase(getNotificationMessages.rejected, (state, { error }) => {
+      state.statusMap.getNotificationMessages = { loading: false, error: error.message };
+    })
+
+    .addCase(getNotificationMessages.fulfilled, (state, { payload: messages }) => {
+      state.messages = messages.filter((message) => {
+        return message.active === true;
+      });
+      state.statusMap.getNotificationMessages = {};
```

Comment:
this should be `initialStatus` now right?

Reply:
it does make sense to set it as initialStatus, since after the request is fulfilled, loading should be false and there shouldn't be an error. 

I noticed in other reducers however that the .fulfilled is set to {}

Reply:
I prefer being more explicit and having `initialStatus ` too, but as you have it also works now since we dont have expressions like `if(getNotificationMessages.loading === false) ...`. But we need to be careful of that.